alf/environments/suite_carla_test.py

```diff
@@ -44,17 +44,19 @@ def test_carla(self):
         logging.info("action_desc: %s" % pprint.pformat(env.action_desc()))
         action_spec = env.action_spec()
 
-        for _ in range(10):
-            action = action_spec.sample([env.batch_size])
-            logging.info("action: %s" % action)
-            action[:, 2] = 0
+        try:
             for _ in range(10):
-                time_step = env.step(action)
-                logging.debug("goal: %s, gnss: %s reward=%s" %
-                              (time_step.observation['goal'][0],
-                               time_step.observation['gnss'][0],
-                               float(time_step.reward[0])))
-        env.close()
+                action = action_spec.sample([env.batch_size])
+                logging.info("action: %s" % action)
+                action[:, 2] = 0
+                for _ in range(10):
+                    time_step = env.step(action)
+                    logging.debug("goal: %s, gnss: %s reward=%s" %
+                                  (time_step.observation['goal'][0],
+                                   time_step.observation['gnss'][0],
+                                   float(time_step.reward[0])))
+        finally:
+            env.close()
 
 
 def play(env):
```

Comment:
Much of this is duplicated with spectator, and doesn't seem like a proper test for the suite_carla?

Reply:
Yes, there are some duplication with spectator. Manual test is still valuable in the sense that you can know what is exactly there by seeing things. Sometime this can help discover problems that automatic test can not reveal